tests/net/lib/mqtt_publisher/src/test_mqtt_publish.c

```diff
@@ -327,96 +326,42 @@ static int test_disconnect(void)
 	return TC_PASS;
 }
 
-static int set_addr(struct sockaddr *sock_addr, const char *addr, u16_t port)
+static int network_setup(void)
 {
-	void *ptr;
+	void *p;
 	int rc;
 
 #ifdef CONFIG_NET_IPV6
-	net_sin6(sock_addr)->sin6_port = htons(port);
-	sock_addr->family = AF_INET6;
-	ptr = &(net_sin6(sock_addr)->sin6_addr);
-	rc = net_addr_pton(AF_INET6, addr, ptr);
+	struct in6_addr addr;
 #else
-	net_sin(sock_addr)->sin_port = htons(port);
-	sock_addr->family = AF_INET;
-	ptr = &(net_sin(sock_addr)->sin_addr);
-	rc = net_addr_pton(AF_INET, addr, ptr);
+	struct in_addr addr;
 #endif
 
-	if (rc) {
-		TC_PRINT("Invalid IP address: %s\n", addr);
-	}
-
-	return rc;
-}
-
-static int network_setup(struct net_context **net_ctx, const char *local_addr,
-		const char *server_addr, u16_t server_port)
-{
 #ifdef CONFIG_NET_IPV6
-	socklen_t addr_len = sizeof(struct sockaddr_in6);
-	sa_family_t family = AF_INET6;
-
-#else
-	socklen_t addr_len = sizeof(struct sockaddr_in);
-	sa_family_t family = AF_INET;
-#endif
-	struct sockaddr server_sock, local_sock;
-	void *p;
-	int rc;
-
-	rc = set_addr(&local_sock, local_addr, 0);
-	if (rc) {
-		TC_PRINT("set_addr (local) error\n");
-		return TC_FAIL;
+	rc = net_addr_pton(AF_INET6, CONFIG_NET_APP_MY_IPV6_ADDR, &addr);
+	if (rc < 0) {
+		goto lb_exit;
 	}
 
-#ifdef CONFIG_NET_IPV6
 	p = net_if_ipv6_addr_add(net_if_get_default(),
-			&net_sin6(&local_sock)->sin6_addr,
-			NET_ADDR_MANUAL, 0);
-#else
-	p = net_if_ipv4_addr_add(net_if_get_default(),
-			&net_sin(&local_sock)->sin_addr,
-			NET_ADDR_MANUAL, 0);
-#endif
-
+				&addr, NET_ADDR_MANUAL, 0);
 	if (!p) {
-		return TC_FAIL;
-	}
-
-	rc = net_context_get(family, SOCK_STREAM, IPPROTO_TCP, net_ctx);
-	if (rc) {
-		TC_PRINT("net_context_get error\n");
-		return TC_FAIL;
-	}
-
-	rc = net_context_bind(*net_ctx, &local_sock, addr_len);
-	if (rc) {
-		TC_PRINT("net_context_bind error\n");
 		goto lb_exit;
 	}
-
-	rc = set_addr(&server_sock, server_addr, server_port);
-	if (rc) {
-		TC_PRINT("set_addr (server) error\n");
+#else
+	rc = net_addr_pton(AF_INET6, CONFIG_NET_APP_MY_IPV4_ADDR, &addr);
```

Comment:
There should be no need to set the IP addresses manually, just call net_app_init() and it will handle all of this setup.